doc/packet_trimming/packet-trimming-design.md

```diff
@@ -0,0 +1,817 @@
+
+**Packet trimming and filtering:**
+```json
+{
+    "ACL_TABLE_TYPE": {
+        "TRIMMING_L3": {
+            "MATCHES": [
+                "SRC_IP"
+            ],
+            "ACTIONS": [
+                "DISABLE_TRIM_ACTION"
+            ],
+            "BIND_POINTS": [
+                "PORT"
+            ]
+        }
+    },
+    "ACL_TABLE": {
+        "TRIM_TABLE": {
+            "POLICY_DESC": "Packet trimming",
+            "TYPE": "TRIMMING_L3",
+            "STAGE": "INGRESS",
+            "PORTS": [
+                "Ethernet0"
+            ]
+        }
+    },
+    "ACL_RULE": {
+        "TRIM_TABLE|TRIM_RULE": {
+            "PRIORITY": "999",
+            "SRC_IP": "1.1.1.1/32",
+            "PACKET_ACTION": "DISABLE_TRIM"
+        }
+    },
+    "BUFFER_POOL": {
+        "egress_lossy_pool": {
+            "mode": "dynamic",
+            "type": "egress"
+        }
+    },
+    "BUFFER_PROFILE": {
+        "q_lossy_trim_profile": {
+            "dynamic_th": "3",
+            "pool": "egress_lossy_pool",
+            "size": "0",
+            "packet_discard_action": "trim"
+        }
+    },
+    "BUFFER_QUEUE": {
+        "Ethernet0|3": {
+            "profile": "q_lossy_trim_profile"
+        }
+    },
+    "SWITCH_TRIMMING": {
+        "GLOBAL": {
+            "size": "128",
+            "dscp_value": "48",
+            "queue_index": "6"
+        }
+    }
+}
+```
+
+### 2.4.5 Initial configuration
+
+No special handling is required: disabled by default
+
+### 2.4.6 Configuration migration
+
+No special handling is required: adding new and extending the existing schema
+
+## 2.5 Flows
+
+### 2.5.1 Config section
+
+### 2.5.1.1 Switch trimming update
+
+![Switch trimming update flow](images/pt_switch_flow.svg "Figure 3: Switch trimming update flow")
+
+###### Figure 3: Switch trimming update flow
+
+### 2.5.1.2 Buffer profile update
+
+![Buffer profile update flow](images/pt_buffer_profile_flow.svg "Figure 4: Buffer profile update flow")
+
+###### Figure 4: Buffer profile update flow
+
+### 2.5.2 Show section
+
+#### 2.5.2.1 Switch trimming show
+
+![Switch trimming show flow](images/pt_show_flow.svg "Figure 5: Switch trimming show flow")
+
+###### Figure 5: Switch trimming show flow
+
+## 2.6 CLI
+
+### 2.6.1 Command structure
+
+**User interface**:
+```
+config
+|--- switch-trimming
+     |--- global [OPTIONS]
+
+show
+|--- switch-trimming
+     |--- global [OPTIONS]
+```
+
+**Options:**
+
+_config switch-trimming global_
+1. `-s|--size` - size (in bytes) to trim eligible packet
+2. `-d|--dscp` - DSCP value assigned to a packet after trimming
+3. `-q|--queue` - queue index to use for transmission of a packet after trimming
+
+_show switch-trimming global_
+1. `-j|--json` - display in JSON format
+
+### 2.6.2 Usage examples
+
+#### 2.6.2.1 Config command group
+
+**The following command updates switch trimming global configuration:**
+```bash
+config switch-trimming global \
+--size 128 \
+--dscp 48 \
+--queue 6
+```
```

Comment:
can we also update mmuconfig CLI to update trim/drop action?

Reply:
@kperumalbfn there is no dedicated CLI to operate on a lossy queue buffer profiles, so no plans to do it

Reply:
@nazariig Currently mmuconfig CLI can be used to update the alpha/dynamic_th. We can use that to enable trim/drop action.

https://github.com/sonic-net/sonic-utilities/blob/ebb17ca86839635275698fff42c38f1575eac838/scripts/mmuconfig#L14

Reply:
@kperumalbfn added